src/main/java/graphql/schema/DataFetchingEnvironmentImpl.java

```diff
@@ -46,15 +48,15 @@ public DataFetchingEnvironmentImpl(Object source,
                                        ExecutionTypeInfo fieldTypeInfo,
                                        ExecutionContext executionContext) {
         this.source = source;
-        this.arguments = arguments;
+        this.arguments = arguments == null ? Collections.emptyMap() : unmodifiableMap(arguments);
```

Comment:
can we instead of having a unmodifiable map, return always a copy in getArguments? I prefer it because it is less surprising for the client imho.

Reply:
It is also the dominant pattern in graphql-java overall I think.